### PR TITLE
Don't iterate over non-enumerable

### PR DESCRIPTION
**Pull request by sroussey**
### What does this PR do?

This changes object property iteration to skip all that have enumeration false. This could have unintended side effects that are wide ranging depending on how the function is used.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Tests not written yet, thus currently a draft.

Related: https://github.com/oven-sh/bun/issues/8316

### CODE REVIEW

File: src/bun.js/bindings/bindings.cpp [hunk 1]
```diff
@@ -4795,7 +4795,11 @@ void JSC__JSValue__forEachProperty(JSC__JSValue JSValue0, JSC__JSGlobalObject* g
         bool anyHits = false;
         JSC::JSObject* objectToUse = prototypeObject.getObject();
         structure->forEachProperty(vm, [&](const PropertyTableEntry& entry) -> bool {
-            if ((entry.attributes() & (PropertyAttribute::Function)) == 0 && (entry.attributes() & (PropertyAttribute::Builtin)) != 0) {
+            int attributes = entry.attributes();
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp]:
```suggestion
            const int attributes = entry.attributes();
```

**Comment by Jarred-Sumner** [comment]
There will be a handful of test failures

**Comment by Jarred-Sumner** [comment]
Would be worth looking at the console-log test failures

**Comment by sroussey** [comment]
I see why it failed. But the more I look the less qualified I feel to fix the enumerable issue.

For example, what is `fast` and why is it there?

In the fast area, when it sees `underscoreProto` it short-circuits but later in the while loop it only does that if it is also marked at non-enumerable.

And that check for PropertyAttribute::DontEnum happens above another check for it later. Why would it not be down in the lower `if` PropertyAttribute::DontEnum area? Are the side effects in the code in-between problematic?

And what is the point of the `prototypeCount`?

_[edit: never mind the questions above, i have a better idea of what's going on now]_

**Comment by sroussey** [comment]
Also, it looks like `JSC__JSValue__forEachPropertyOrdered` has more different with it than `JSC__JSValue__forEachProperty` that goes beyond ordering.

**Comment by jdalton** [comment]
Related to https://github.com/oven-sh/bun/pull/6654

**Comment by sroussey** [comment]
I guess my test should move to the main console.log test. Now that I look at it, it is under bun folder which I now see how bun specific tests, which this is not.

**Comment by sroussey** [comment]
I guess I can't get away from coding on console.log. Another decade... lol

**Comment by sroussey** [comment]
BTW: any reason `@actions/core` is not installed? It is required by the tests.

**Pull request closed without merging**